### PR TITLE
skip smudge by default and provide server_options override

### LINKED ISSUE #74: Do not load LFS objects by default
We should have `GIT_LFS_SKIP_SMUDGE=1` set by default and the possibility to override it via `server_options`.

### PR DESCRIPTION
**Pull request by rokroskar**
closes #74

### CODE REVIEW

**Comment by rokroskar** [comment]
Yep you're absolutely right - we could just make the default that nothing gets pulled from LFS.

File: helm-chart/renku-notebooks/values.yaml [hunk 1]
```diff
@@ -101,6 +101,10 @@ serverOptions:
   defaultUrl:
     default: /lab
     options: [/lab]
+  lfs_skip_smudge:
```
Comment (ableuler) [inline helm-chart/renku-notebooks/values.yaml]: These changes won't be picked up by the UI. So far I assumed that the top-level properties are only `defaultUrl` and `resources`, and I display all options under `resources` automatically. If you really want this to be a configurable option through the UI, add the `enum` type to the `lsf_skip_smudge` object and move it into `resources`. In general I would prefer a flat structure for the `server_options` object as long as the total number of options is manageable.

Comment (rokroskar) [reply helm-chart/renku-notebooks/values.yaml]: what about 

```
serverOptions:
  cpu_request:
    displayName: Number of CPUs
    type: integer
    default: 1
    range: [1, 8]
  mem_request:
    displayName: Memory
    type: enum
    default: 2G
    options: [2G, 4G, 8G]
  gpu:
    displayName: Number of GPUs
    type: integer
    default: 0
    range: [0, 4]
  defaultUrl:
    default: /lab
    options: [/lab]
  lfs_skip_smudge:
    default: 1
    options: [0,1]
    displayName: Do not automatically fetch data
```

Comment (ableuler) [reply helm-chart/renku-notebooks/values.yaml]: Yes, this would make the ui a bit simpler.

Comment (rokroskar) [reply helm-chart/renku-notebooks/values.yaml]: I had `resources` in there initially to be a bit closer to the k8s pod spec, but I need to do the mapping by hand anyway so it's not very useful.

File: helm-chart/renku-notebooks/values.yaml [hunk 2]
```diff
@@ -82,25 +82,30 @@ tolerations: []
...
+  cpu_request:
+    displayName: CPU request
+    type: enum
+    default: 0.1
+    options: [0.1, 0.5, 1.0, 2.0, 4.0]
+  mem_request:
+    displayName: Memory request
+    type: enum
+    default: 1G
+    options: [1G, 2G, 4G]
```
Comment (leafty) [inline helm-chart/renku-notebooks/values.yaml]: `1G` or `1Gi`? I like my power of teo :wink:

File: helm-chart/renku-notebooks/values.yaml [hunk 3]
```diff
@@ -82,25 +82,30 @@ tolerations: []
...
+  gpu_request:
+    displayName: GPU request
+    type: int
+    default: 0
+    range: [0, 4]
```
Comment (leafty) [inline helm-chart/renku-notebooks/values.yaml]: Maybe have `[0 , 1]` as the default range?

File: src/notebooks_service.py [hunk 4]
```diff
@@ -133,8 +133,8 @@ def authenticated(f):
     def decorated(*args, **kwargs):
         token = request.cookies.get(
             auth.cookie_name
-        ) or request.headers.get('Authorization',
-                                 'token').split('token', 1)[1].strip()
+        ) or request.headers.get('Authorization', 'token').split('token',
+                                                                 1)[1].strip()
```
Comment (leafty) [inline src/notebooks_service.py]: Will fail on headers like `Authorization: ehskdfbklshfuiw` with http 500. We can merge and create an issue just after.

File: helm-chart/renku-notebooks/values.yaml (hunk 3 above)
Comment (leafty) [reply helm-chart/renku-notebooks/values.yaml]:
```suggestion
    range: [0, 1]
```

File: helm-chart/renku-notebooks/values.yaml [hunk 5]
```diff
@@ -99,7 +99,7 @@ serverOptions:
     range: [0, 1]
   defaultUrl:
     displayName: default landing URL of the notebook server
-    type: string
+    type: enum
```
Comment (leafty) [inline helm-chart/renku-notebooks/values.yaml]: why? what if someone has a new thing like r shiny?

Comment (ableuler) [reply helm-chart/renku-notebooks/values.yaml]: We can define a "string" option (not implemented yet on the ui side), but then the options don't make sense. The original plan was to provide options that would translate into a dropdown.

Comment (rokroskar) [reply helm-chart/renku-notebooks/values.yaml]: `enum` is correct, this was my mistake

**Pull request merged**